text/0001-catalogs.md

```diff
@@ -0,0 +1,346 @@
+
+For example, the `foo`, `bar`, and `baz` workspace packages may all be using a catalog version for `react` on `^18.2.0`. However, `baz` may depend on `quz-react-components`, which brings in an older version of React.
+
+```mermaid
+flowchart LR
+  classDef project color:#333333,fill:#bdd5fc,stroke:#333,stroke-width:1px
+  classDef package color:#333333,fill:#e5fce7,stroke:#05a000,stroke-width:1px
+  classDef node_modules color:#333333,fill:#f5f5f5,stroke:#999,stroke-width:1px
+
+  foo:::project
+  bar:::project
+  baz:::project
+
+  node_modules:::node_modules
+
+  foo --> react
+  bar --> react
+  baz --> react
+
+  baz --> quz-react-components --> react16
+
+  subgraph node_modules
+    react["react@18.2.0"]:::package
+    quz-react-components["quz-react-components@0.1.0"]:::package
+    react16["react@16.14.0"]:::package
+  end
+```
+
+This scenario requires human intervention and is outside of the scope of this RFC. It's may be important to note that Yarn allows users to detect these sorts of situations and fail installation through a mechanism called [Constraints](https://yarnpkg.com/features/constraints).
+
+## Alternatives
+
+### Syncpack
+
+[Syncpack](https://github.com/JamieMason/syncpack/) is a great open source tool for keeping `package.json` dependency specifiers in sync on disk.
+
+The proposed solution allows metadata to be defined in a singular file without copying definitions to other files on disk. This is a capability only possible by the package manager reading `package.json` files.
+
+### Comparison to overrides/resolutions
+
+An alternative mechanism for the version catalog is the [`pnpm.overrides` feature](https://pnpm.io/package_json#pnpmoverrides). While mechanically this allows you to set the version of a dependency across all workspace packages, it can be a bit unexpected when if `pnpm.overrides` rewrites a dependency's dependency to an incompatible version silently.
+
+`pnpm.overrides` is ultimately intended for a different purpose. The NPM RFC for a similar feature explicitly states that it should be used as a short-term hack to fix vendor problems.
+
+> Using this feature should be considered a hack in most cases, something that is done temporarily while waiting for a bug to be fixed, or to avoid excessive duplication caused by an overly strict meta-dependency specifier.
+https://github.com/npm/rfcs/blob/main/accepted/0036-overrides.md
+
+The `catalog:` protocol is conversely intended for long-lived usage.
+
+## Prior Art
+
+- The Gradle build tool has a [similar concept called _version catalogs_](https://docs.gradle.org/current/userguide/platforms.html#sub:central-declaration-of-dependencies). This RFC's name was [directly chosen as inspiration](https://github.com/pnpm/pnpm/issues/2713#issuecomment-1341476866).
+- Syncpack as mentioned above in the [alternatives section](#syncpack).
+- Rush has a ["Preferred versions" concept](https://rushjs.io/pages/advanced/preferred_versions/) that works similarly.
```

Comment:
Rust also has a mechanism to inherit package version from the so-called package table:
* https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace
* https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

Reply:
Added in commit cefcf14b483d591c6124e3b853f4532828e460d9.